src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -338,9 +337,11 @@ private Task StartAsyncCore(TransferFormat transferFormat)
                         Log.ErrorDuringClosedEvent(_logger, ex);
                     }
 
+                    // signal that close has completed
+                    _closeTcs.SetResult(null);
```

Comment:
The `Closed` event was running after this close TCS was being set as complete, which was causing `HubConnectionTests.CanStopAndStartConnection` to fail because a connection was being started up again while it was still shutting down by `Closed`.